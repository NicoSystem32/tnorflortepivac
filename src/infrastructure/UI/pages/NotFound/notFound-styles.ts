import { styled } from 'twin.macro'

export const NotFoundContent = styled.div`
    @media (min-width: 1180px) and (max-width: 1400px) {
        display: grid;
        grid-template-columns: 53% 45%;
        grid-template-areas: 'info404 img404';
    }

    @media (min-width: 1400px) {
        display: grid;
        grid-template-columns: 52% 50%;
        grid-template-areas: 'info404 img404';
        margin: 3% 15%;
    }
`
export const NotFoundContentImg = styled.div`
    width: 80%;
    margin: 5% 10%;
    background-image: url(../../assets/icons/big-stain.png);
    background-repeat: no-repeat;
    background-size: contain;
    grid-area: img404;
    background-position: bottom;

    .img-not-found {
        width: 100%;
    }

    @media (min-width: 570px) and (max-width: 1180px) {
        background-size: 80%;

        .img-not-found {
            width: 80%;
        }
    }

    @media (min-width: 1180px) and (max-width: 1400px) {
        margin: 5% 0% 0 0;
        padding: 0 22% 8% 0;
        background-size: 345px;
        background-position: right;

        .img-not-found {
            width: 115%;
        }
    }

    @media (min-width: 1400px) {
        margin: 5% 0% 0 0;
        padding: 0 5% 16% 0;
        background-size: 510px;
        background-position: right;

        .img-not-found {
            width: 80%;
        }
    }
`
export const NotFoundContentInfo = styled.div`
    background-image: url(../../assets/icons/small-stain.png);
    background-repeat: no-repeat;
    background-position: center;
    background-size: 65%;
    width: 90%;
    margin: 0 5%;
    padding: 10% 0 15% 0;
    grid-area: info404;

    .title-404 {
        font-size: 125px;
        font-family: var(--font-main-bold);
        color: var(--text-opacity9);
        margin: auto;

        span {
            color: var(--dominant-color-dark) !important;
        }
    }

    subtitle-404 {
        font-size: 14px;
        font-family: var(--font-secundary-regular);
        color: var(--text-opacity9);
        strong {
            font-family: var(--font-secundary-bold);
        }
    }

    .content-position {
        text-align: center;
    }

    @media (min-width: 570px) and (max-width: 1180px) {
        width: 70%;
        margin: 0 15%;
    }

    @media (min-width: 1180px) and (max-width: 1400px) {
        background-position: left;
        background-size: 40%;
        padding: 10% 24% 15% 0%;
        margin-left: 25%;

        .content-position {
            text-align: start;
        }

        subtitle-404 {
            font-size: 16px;
        }
    }

    @media (min-width: 1400px) {
        background-position: left;
        background-size: 40%;
        padding: 10% 24% 15% 0%;
        margin-left: 25%;

        .content-position {
            text-align: start;
        }

        subtitle-404 {
            font-size: 16px;
        }
    }
`
