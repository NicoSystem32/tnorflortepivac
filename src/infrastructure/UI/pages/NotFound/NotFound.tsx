import { useHistory } from 'react-router-dom'
import { Button } from '../../components'

// Icons
import { Big404SVG } from '../../utils/getIcons'

// Styles
import { NotFoundContent, NotFoundContentImg, NotFoundContentInfo } from './notFound-styles'

const NotFound = (): JSX.Element => {
    // initial declarations
    const history = useHistory()

    const redirections = (url: string): void => {
        history.push(url)
    }

    return (
        <NotFoundContent>
            <NotFoundContentImg>
                <img className="img-not-found" src={Big404SVG} alt="Not found" />
            </NotFoundContentImg>
            <NotFoundContentInfo>
                <img src="" alt="" />
                <p className="title-404 content-position">
                    4<span className="text-yellow">0</span>4
                </p>
                <p className="subtitle-404">
                    Parece que esta página ahorró lo suficiente y se fue de vacaciones,
                    <strong>
                        {' '}
                        puede que esta página este en mantenimiento o haya sido eliminada,
                    </strong>{' '}
                    regresa al home e intenta mas tarde.
                </p>
                <div className="content-position">
                    <Button variant="sub-dominant" onClick={() => redirections('/home-wallet')}>
                        Volver al home
                    </Button>
                </div>
            </NotFoundContentInfo>
        </NotFoundContent>
    )
}
export default NotFound
