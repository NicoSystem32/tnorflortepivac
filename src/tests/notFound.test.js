import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import NotFound from '../infrastructure/UI/components/NotFound/notFound'

describe('Pruebas en <NotFound />', () => {
    const wrapper = shallow(<NotFound />)

    test('debe de mostrar <NotFound /> correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
