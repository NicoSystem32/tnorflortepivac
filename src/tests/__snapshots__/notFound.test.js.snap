// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NotFound /> debe de mostrar <NotFound /> correctamente 1`] = `
<div
  className="content-404"
>
  <div
    className="content-img-404"
  >
    <img
      alt=""
      className="img-404"
      src="big-404.png"
      srcSet="big-404.svg"
    />
  </div>
  <div
    className="content-info-404"
  >
    <img
      alt=""
      src=""
    />
    <p
      className="text-biger-type content-run-404"
    >
      4
      <span
        className="text-yellow"
      >
        0
      </span>
      4
    </p>
    <p
      className="text-secundari-type"
    >
      Parece que esta página ahorró lo suficiente y se fue de vacaciones,
      <strong>
         puede que esta página este en mantenimiento o haya sido eliminada,
      </strong>
       regresa al home e intenta mas tarde.
    </p>
    <div
      className="content-run-404"
    >
      <Button
        active={false}
        className="btn-redirection btn-global"
        disabled={false}
        onClick={[Function]}
        variant="primary"
      >
        Volver al home
      </Button>
    </div>
  </div>
</div>
`;
